### DISCUSSION TITLE
Simplification Options

### DISCUSSION THREAD

**Discussion by mattpow35** [Whinarn/UnityMeshSimplifier#55] [Q&A] [answered]
What is the correct way to change the simplification options in code?? does anyone have any examples they could share that would be much appreciated.

**Comment by Whinarn** [comment, marked as answer]
The [SimplificationOptions](https://github.com/Whinarn/UnityMeshSimplifier/wiki/Mesh-Simplifier-API#simplificationoptions) is a struct that can be changed through the `SimplificationOptions` property of the `MeshSimplifier` class (see list of properties [here](https://github.com/Whinarn/UnityMeshSimplifier/wiki/Mesh-Simplifier-API#properties)). Or you can use them as an input for the [LOD Generator API](https://github.com/Whinarn/UnityMeshSimplifier/wiki/LOD-Generator-API).

You can create the struct very easily, with your desired properties like this:
```c#
var simplificationOptions = new SimplificationOptions
{
    PreserveBorderEdges = false,
    PreserveUVSeamEdges = false,
    PreserveUVFoldoverEdges = false,
    PreserveSurfaceCurvature = false,
    EnableSmartLink = true,
    VertexLinkDistance = double.Epsilon,
    MaxIterationCount = 100,
    Agressiveness = 7.0,
    ManualUVComponentCount = false,
    UVComponentCount = 2
};

var meshSimplifier = new MeshSimplifier();
meshSimplifier.SimplificationOptions = simplificationOptions;
meshSimplifier.Initialize(mesh);
meshSimplifier.SimplifyMesh(quality);
```

Or you can start with the default options and make modifications:
```c#
var simplificationOptions = SimplificationOptions.Default;
simplificationOptions.EnableSmartLink = false;
```

Is this what you had in mind?

Please see the very fresh documentation on the [wiki](https://github.com/Whinarn/UnityMeshSimplifier/wiki) as well, which could hopefully clear up a few things.

**Reply by mattpow35** [reply]
Yes! This is what I was looking for thank you!

**Reply by Mrchs** [reply]
I changed the original PreserveBorderEdges setting to true so that my model preserves the boundaries while simplifying the grid, or does the default false include this feature, but I have a lot of holes by default
